### ISSUE TITLE
Coprorate repositories in Azure DevOps cannot be cloned

### ISSUE THREAD

**Issue by PowellDean** [gitbutlerapp/gitbutler#5848]
I am trying to clone a repository stored in my company's Azure DevOps subscription. I receive the following error message:

Error
Error: The object <long random string of characters snipped> could not be decoded or wasn't found.

I receive the same error trying both SSH and HTTPS addresses. I am able to clone my personal GitHub and Codeberg reppositories with no issue.

Is this an Azure DevOps security issue?

**Comment by Byron** [comment]
Thanks a lot for reporting!

It's a known issue with the way `gitoxide`, the crate that clones repositories, handles packs that "shouldn't be". All the details are [in the tracking issue](https://github.com/GitoxideLabs/gitoxide/issues/1025), but in short, a fix means to provide an entirely new and in all likelihood slower implementation of cloning which is more similar to the one Git uses.

> Is this an Azure DevOps security issue?

Definitely not.

**Comment by PowellDean** [comment]
Thanks for the quick reply. I completely understand the "probably won't fix" rationale given in the linked issue. I am also aware the workaround is to just clone the repository locally and import it (which I can confirm did work). Unfortunately, in a corporate setting such as the one I'm in, the word "workaround" for a fundamental function like cloning a remote will kill any chance I have of introducing the tool.

You can close the issue if you like. I won't be able to proceed any further with test-driving the application.

It's an interesting tool though. Thanks for the opportunity to look at it.

Best Regards,
Dean Powell
Edmonton, Canada

**Comment by Byron** [comment]
Thanks for the kind words! 

The issue should stay open as I believe it will be fixed eventually :).

**Comment by Caleb-T-Owens** [comment]
🤔 We could move from gitoxide to shelling out to the git executable like we do with fetch and push. It would allow us to also make use of askpass for SSH keys with passwords

**Comment by Byron** [comment]
> 🤔 We could move from gitoxide to shelling out to the git executable like we do with fetch and push.

Yes, definitely. Since the current implementation also doesn't use progress or can be interrupted, it's not very suited for huge repositories to be cloned, so the performance benefit of `gitoxide` over Git won't matter much either.

> It would allow us to also make use of askpass for SSH keys with passwords

`gitoxide` also uses `ssh` (the program) as transport and I'd think that credential handling associated with it works similarly. But I wouldn't bet on it either as I haven't tested that specifically, nor did I test more advanced setups.

**Comment by Caleb-T-Owens** [comment]
I'll add it to my todo list for the new year, unless @krlvi wants to take a look at this himself sooner given that it's very much onboarding related?

**Comment by PowellDean** [comment]
I can help with any testing. Our repositories are usually quite small, but
we have a lot of them spread across several AZDO instances.

Thanks, all!

Dean

**Comment by krlvi** [comment]
Hey @PowellDean, sorry this took so long. This was fixed in https://github.com/gitbutlerapp/gitbutler/pull/11434 — GitButler now uses the system Git executable for cloning instead of gitoxide, which removes the Azure DevOps pack incompatibility you hit. The change shipped in [0.18.3](https://github.com/gitbutlerapp/gitbutler/releases/tag/release/0.18.3).

I'm closing this as fixed. If Azure DevOps cloning still fails on a current release, please open a fresh issue with the error and platform.
